### ISSUE TITLE
Aragon Futarchy App

### ISSUE THREAD

**Issue by cwhinfrey** [aragon/nest#93]
# Aragon Futarchy App:

## Abstract

Futarchy is an alternative and relatively untested form of governance that uses prediction markets to determine which policies will be the most beneficial for an organization. Giving Aragon DAOs access to futarchy creates a low stakes and low barriers-to-entry environment for continued testing and experimentation. While there are many potential implementations of futarchy, we propose starting with a single implementation described below.

#### Futarchy Setup

- __Success Metric__ - This implementation will use token price to predict and measure the success of a policy or decision. A price feed oracle for the DAO's token will be required when the Aragon app is added. If there is demand for other success metrics in the future, they can be supported fairly easily as long as an oracle can be provided.

- __Decision Function__ - The decision will be made based on the outcome that has the higher predicted token value for the largest percentage of the decision period.

- __Tokenized Event Configuration__ - For each futarchy decision, a categorical event for the decision being accepted or not accepted will be deployed. The outcome tokens of the categorical event will be used as the collateral tokens for two scalar events that will be used to predict the token price given that outcome. For more information on this setup, please see the first configuration described [here](https://ethresear.ch/t/possible-futarchy-setups/1820).

- __LMSR Markets__ - For each scalar event, an LMSR market will be started for the trading of the scalar event's outcome tokens. LMSR markets require up front funding to ensure a sufficiently liquid market. The burden of funding the market will lie on the user that creates the futarchy decision.

#### Integration With Aragon

The futarchy app will follow the patterns set by the voting app as closely as possible. Ideally, DAOs will be able to swap out voting for futarchy with minimal effort. The proposed interface for the Aragon futarchy app can be found [here](https://gist.github.com/cwhinfrey/0dc3e4750919bbbbb8f6cd8b7ea4dcc4).

A UI will be designed and implemented to allow end users to launch new futarchy decisions, interact with markets, and gain insights on active futarchy decisions. It will consist of a screen outlining all current and past futarchy decisions, a detail screen for specific futarchy decisions, and a new futarchy decision slide out modal similar to the new vote modal.

## Prior Code and UI

We will use the futarchy setup from https://github.com/levelkdev/fcr-contracts (Futarchy Curated Registry project collaboration with the Gnosis team) and the pieces of the https://github.com/levelkdev/fcr-js JS library that are related to interaction with futarchy smart contracts.

The Gnosis team also has some futarchy trading UI design mocks that may be applicable to this project.

## Deliverables and Timeline

Aragon Futarchy app deployed on testnet, with a user interface for creating and interacting with futarchy markets.

Estimated total timeframe of 3 months for project completion of these features:

| Expected Timeframe | Feature Description |
| --- | --- |
| month 1 | UI Mocks for futarchy app view |
| month 1 | Finalize Futarchy app + unit tests |
| month 2 | Integration tests for launching, running, and resolving futarchy decisions |
| month 2 | Smart contracts(s) for LMSR bounds oracle + tests |
| month 2 | Smart contract(s) for PriceFeedOracle + tests |
| month 1-2 | Build web UI w/ mock data |
| month 2-3 | Integrate web UI w/ smart contracts |

## Grant size

Funding: $100k in ETH, split into chunks paid out over achieved deliverables.

Success reward: Up to $50k in ANT, given out when all deliverables are ready.

**Comment by mariapao** [comment]
Hi @cwhinfrey we have reviewed the proposal and we can't wait to have this app on Aragon to start experimenting with this governance model. Also, you guys have a very strong team. Therefore, we have decided to approve the proposal!

The second part of the application is the request for funding. [Here](https://github.com/aragon/nest/blob/master/Guide_for_submitting_a_request_for_funding.md) is a guide to submit such request for funding. In this stage of the application we review in detail the PoC/whitepaper, roadmap, deliverables and the time commitment of the team. 

We look forward to reviewing the second part of the application :)
